### PR TITLE
[INFRA] add workflow to update submodule automatically

### LINKED ISSUE #1450: add CI workflow that regularly updates git submodules
https://github.com/bids-standard/bids-validator/blob/b616bd19feb15baf14ec3e8b292f147213a34e46/.gitmodules#L1-L3

based on a point raised by @Remi-Gau

### PR DESCRIPTION
**Pull request by Remi-Gau**
closes #1450

### CODE REVIEW

**Comment by Remi-Gau** [comment]
Added a lot of comments in the yml file. Feel free to remove.

File: .github/workflows/update_submodules.yml [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+# This workflow will checkout the tracked branch of all (or a subset of) submodules
+# and will update the submodule pull the latest commit of that submodule branch 
+
+# If the branch was not up to date this will then open a PR against the base branch
+# in the repo this submodule belongs to.
+
+# ----  WARNING ----
+# Currently if the submodule is not tracking the default branch, 
+# this will workflow will fail.
```
Comment (Remi-Gau) [inline .github/workflows/update_submodules.yml:11]: This is currently the main limitation of this workflow.

Been trying to figure it out and asked help from our favorite submodules expert (the datalad team) but could not figure it out.

**Review by sappelhoff — APPROVED** [review]
nice! Thanks Remi!

**Comment by sappelhoff** [comment]
interesting, this fails without informative logs now on master :thinking: https://github.com/bids-standard/bids-validator/runs/6914179912?check_suite_focus=true

**Comment by Remi-Gau** [comment]
Sigh...
Here we go for another round of "FCI' commits. (obviously this stands for `F... ix Continuous Integration`)
OK let me debug this on my fork to fix.

**Comment by Remi-Gau** [comment]
Now it works... on my fork.

- https://github.com/Remi-Gau/bids-validator/actions/runs/2507965359
- see opened PR: https://github.com/Remi-Gau/bids-validator/pulls

Will open a fix here. #1469

**Pull request merged**